msg/Detection2D.msg

```diff
@@ -19,7 +19,8 @@ sensor_msgs/Image source_img
 # If true, this message contains object tracking information.
 bool is_tracking
 
-# ID used for consistency across multiple detection messages. This value will
-#   likely differ from the id field set in each individual ObjectHypothesis.
-# If you set this field, be sure to also set is_tracking to True.
-string tracking_id
+# ID used for consistency across multiple detection messages. Detections
```

Comment:
I don't think this is the best name, if this were to be a `uuid`, `object_id` doesn't exactly align with that, since the word "object" implies something like naming the class or something. I agree it shouldn't be `tracking_id`, but `id` would be OK too. its not too uncommon for the same field to exist in multiple messages with different contexts. 

If we changed Detection*D to `type` rather than `id` and then this could be `id` would work too.

Reply:
I think of objects and classes here the same way as in programming languages (class = "human", object = "Steve"). Actually I would be happy with leaving this as `tracking_id`, since that's essentially what it is. It remains constant as long as it's tracked, then when the object leaves the field of view and comes back out probably gets a new id. I only included this change because I knew you'd insist. The only problem with leaving this as `tracking_id` is that it would be confusing once we add a proper tracking message (with velocities etc.). So yeah, `id` would work for me, and it's better than `object_id`.

Reply:
> I think of objects and classes here the same way as in programming languages

I don't think you should. This is `vision_msgs` and those words have meaning in a vision context. We should use that nomenclature. 

So `id` it is then?

Reply:
Oh, I'm perfectly aware of the vision nomenclature, I happen to work in that area. Care to explain how "the word "object" implies something like naming the class or something" is standard vision nomenclature, and how they don't differentiate between objects and classes?

Reply:
Might I suggest `instance_id`? I could see this working for both UUIDs or strings.

Reply:
Ok, I've updated the PR to use `instance_id`. @SteveMacenski : What do you think?

Reply:
`instance_id` I do think is better. But I still think just plain `id` makes this sufficiently general that it is semantically clear can be an instance id like `human32542` or a UUID. I think the word `instance` puts your hand on the scale a little for how people will read this.

Reply:
ah, yes, I think this goes back to my incomplete understanding of how you propose to use UUIDs. If you feel that `instance_id` and UUIDs conflict, then we should keep it as neutral `id` for now. We can always update it later - and in fact, if we switch to UUIDs, we probably need to update the comment anyway. 

The only thing I don't like about this is that the PR now removes one field called `id`, and creates a field on a different message that is called `id`. It could be confusing for migration purposes. But I think it's OK--since we are already changing the package significantly, no reason to hold back now :)

Reply:
We're changing a few other things and it will be changed between distributions so I think this is fine. Plus they're different messages so compilation will fail and it should be clear.

@mintar `id` acceptable to you?

Reply:
As requested, I've split the PR into two separate commits and renamed `instance_id` (formerly known as `tracking_id`) to `id`.